cli/impl/src/test/java/io/helidon/build/cli/impl/CommandInvoker.java

```diff
@@ -391,9 +398,28 @@ public InvocationResult invokeInit() throws Exception {
             args.forEach(a -> System.out.print(a + " "));
             System.out.println();
 
-            // Execute and verify process exit code
-            String output = TestUtils.execWithDirAndInput(workDir.toFile(), input, argsArray);
-            return new InvocationResult(this, output);
+            return new InvocationResult(this, execute(workDir.toFile(), input, argsArray));
+        }
+
+        private String execute(File wd, File input, String... args) throws Exception {
+            if (Stream.of(execScript, execHelidonClass, execNativeImage).filter(b -> b).count() > 1L) {
+                throw new IllegalStateException("Several executable cannot be run in the same process");
+            }
+
+            if (execScript) {
+                System.out.println("Executing CLI with helidon script");
+                return TestUtils.execScript(wd, input, args);
```

Comment:
Both new methods `TestUtils.execScript` and `TestUtils.execNativeImage` rely on a system property to find the executable. This is wrong ; `CommandInvoker` implements a builder pattern let it be passed the executable to use explicitly.